modules/src/ics03_connection/exported.rs

```diff
@@ -0,0 +1,113 @@
+use super::error;
+use crate::ics23_commitment::CommitmentPrefix;
+use anomaly::fail;
+use serde_derive::{Deserialize, Serialize};
+
+pub trait ConnectionI {
+    type ValidationError: std::error::Error;
+
+    fn state(&self) -> State;
+    fn client_id(&self) -> String;
+    fn counterparty(&self) -> Box<dyn CounterpartyI<ValidationError = super::error::Error>>;
+    fn versions(&self) -> Vec<String>;
+    fn validate_basic(&self) -> Result<(), Self::ValidationError>;
+}
+
+pub trait CounterpartyI {
```

Comment:
`I` suffix for traits is not a common Rust practice in my experience. How about naming the trait `Counterparty` instead?

Ambiguities can always be resolver by renaming imports or using the qualified path to the trait.